control-plane/pkg/reconciler/broker/broker.go

```diff
@@ -373,7 +373,14 @@ func (r *Reconciler) finalizeKind(ctx context.Context, broker *eventing.Broker)
 
 		// get security option for Sarama with secret info in it
 		securityOption := security.NewSaramaSecurityOptionFromSecret(secret)
-		if err := r.finalizeNonExternalBrokerTopic(broker, securityOption, topicConfig, logger); err != nil {
+		err = r.finalizeNonExternalBrokerTopic(broker, securityOption, topicConfig, logger)
+
+		// if finalizeNonExternalBrokerTopic returns error that kafka is not reachable,
+		// we return nil, and clean up the broker
+		if err != nil {
+			if strings.Contains(err.Error(), "cannot obtain Kafka") {
```

Comment:
Is this error we check for enough?

Reply:
When I check what `finalizeNonExternalBrokerTopic` does, I see it returns an error when:
- It cannot build Sarama config (e.g. because of corrupt configuration, deleted configmap, etc)
- It cannot connect to Kafka
- It cannot delete the topic

IMO, we should simply continue with the deletion of the broker resource in all of those 3 cases. There's not much we can do in none of these cases.

Or, if you want to treat for example case#1 differently, then move that case in this method and return the error.

Reply:
@pierDipi https://github.com/knative-sandbox/eventing-kafka-broker/pull/2570#issuecomment-1234859223

How to track how many retries we made? In status?

My suggestion is to do this retries in a separate PR though.

Reply:
ok, I can change that - I wanted to be more specific, but I see your point. I will update later

thanks

Reply:
I think I disagree. 

I think we should do the retires if we return `nil` on any error. The remote kafka not reachable is IMO a different beast than topic not being able to deleted (see the test-failures when I changed)

Reply:
Ok, sounds good to me.

As we talked in the videochat, we should create multiple functions out of this function and then things would make sense.